spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/SshUri.java

```diff
@@ -37,6 +37,8 @@
 	private String preferredAuthentications;
 	private boolean ignoreLocalSshSettings;
 	private boolean strictHostKeyChecking = true;
+	private String proxyHost;
+	private int proxyPort = 8080;
```

Comment:
Is this a good idea? Seems like this should be a required field, shouldn't it?

Reply:
Do you mean the proxyPort field?

Reply:
Yes, the line I commented on.

Reply:
Ok, I could write a validator which verifies, that proxyPort has to be set in case proxyHost is set.

Reply:
Just do it in `PropertyBasedSshSessionFactory.java` after the not null check for the host. Use spring `Assert` class.